Bottom: 9035596dfb9bd9fc0099851f7680e8441f4025cc
Top:    f34005fa7168db7ee71b00904c43149612a9fc05
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-11 18:11:23 +0000

Refresh of tmp

---

diff --git a/src/draw.c b/src/draw.c
index 5057d54..f2d843e 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1534,7 +1534,7 @@ draw_element (ElementType *element)
  */
 
 void
-hid_expose_callback (HID *expose_hid, HID_DRAW *expose_hid_draw, BoxType *region, void *item)
+hid_expose_callback (HID_DRAW *expose_hid_draw, BoxType *region, void *item)
 {
   hid_draw = expose_hid_draw;
   Output.fgGC = hid_draw_make_gc (hid_draw);
diff --git a/src/hid.h b/src/hid.h
index 12959af..41b24e7 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -548,7 +548,7 @@ typedef enum
    Do *not* assume that the hid that is passed is the GUI hid.  This
    callback is also used for printing and exporting. */
   struct BoxType;
-  void hid_expose_callback (HID *hid_, HID_DRAW *hid_draw, struct BoxType *region_, void *item_);
+  void hid_expose_callback (HID_DRAW *hid_draw, struct BoxType *region_, void *item_);
 
 /* This is initially set to a "no-gui" gui, and later reset by
    main. hid_expose_callback also temporarily set it for drawing. */
diff --git a/src/hid/common/extents.c b/src/hid/common/extents.c
index 20a27d7..d652c1c 100644
--- a/src/hid/common/extents.c
+++ b/src/hid/common/extents.c
@@ -227,7 +227,7 @@ hid_get_extents (void *item)
   region.Y1 = -COORD_MAX - 1;
   region.X2 = COORD_MAX;
   region.Y2 = COORD_MAX;
-  hid_expose_callback (&extents_hid, &extents_graphics, &region, item);
+  hid_expose_callback (&extents_graphics, &region, item);
 
   return &box;
 }
diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index 14f4a1a..d53c491 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -470,7 +470,7 @@ gcode_start_png_export ()
   linewidth = -1;
   lastbrush = (gdImagePtr)((void *) -1);
 
-  hid_expose_callback (&gcode_hid, &gcode_graphics, &region, 0);
+  hid_expose_callback (&gcode_graphics, &region, 0);
 }
 
 static FILE *
diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index b2dbf78..9e6209b 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -683,11 +683,11 @@ gerber_do_export (HID_Attr_Val * options)
   lastgroup = -1;
   layer_list_idx = 0;
   finding_apertures = 1;
-  hid_expose_callback (&gerber_hid, &gerber_graphics, &region, 0);
+  hid_expose_callback (&gerber_graphics, &region, 0);
 
   layer_list_idx = 0;
   finding_apertures = 0;
-  hid_expose_callback (&gerber_hid, &gerber_graphics, &region, 0);
+  hid_expose_callback (&gerber_graphics, &region, 0);
 
   memcpy (LayerStack, saved_layer_stack, sizeof (LayerStack));
 
diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index c745cbe..705b45b 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -801,7 +801,7 @@ redraw_region (GdkRectangle *rect)
 
   ghid_draw_bg_image();
 
-  hid_expose_callback (&ghid_hid, &ghid_graphics, &region, 0);
+  hid_expose_callback (&ghid_graphics, &region, 0);
   ghid_draw_grid ();
 
   /* In some cases we are called with the crosshair still off */
@@ -1179,7 +1179,7 @@ ghid_pinout_preview_expose (GtkWidget *widget,
                       0, 0, allocation.width, allocation.height);
 
   /* call the drawing routine */
-  hid_expose_callback (&ghid_hid, &ghid_graphics, NULL, pinout->element);
+  hid_expose_callback (&ghid_graphics, NULL, pinout->element);
 
   gport->drawable = save_drawable;
   gport->view = save_view;
@@ -1236,7 +1236,7 @@ ghid_render_pixmap (int cx, int cy, double zoom, int width, int height, int dept
   region.Y1 = MAX (0, MIN (PCB->MaxHeight, region.Y1));
   region.Y2 = MAX (0, MIN (PCB->MaxHeight, region.Y2));
 
-  hid_expose_callback (&ghid_hid, &ghid_graphics, &region, NULL);
+  hid_expose_callback (&ghid_graphics, &region, NULL);
 
   gport->drawable = save_drawable;
   gport->view = save_view;
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index b0e7f42..dcd78a1 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1039,7 +1039,7 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
   ghid_draw_bg_image ();
 
   ghid_invalidate_current_gc ();
-  hid_expose_callback (&ghid_hid, &ghid_graphics, &region, 0);
+  hid_expose_callback (&ghid_graphics, &region, 0);
   hidgl_flush_triangles (priv->hidgl);
 
   ghid_draw_grid (priv->crosshair_gc, &region);
@@ -1193,7 +1193,7 @@ ghid_pinout_preview_expose (GtkWidget *widget,
                 gport->view.flip_y ? gport->view.y0 - PCB->MaxHeight :
                                     -gport->view.y0, 0);
 
-  hid_expose_callback (&ghid_hid, &ghid_graphics, NULL, pinout->element);
+  hid_expose_callback (&ghid_graphics, NULL, pinout->element);
   hidgl_flush_triangles (priv->hidgl);
   glPopMatrix ();
 
@@ -1307,7 +1307,7 @@ ghid_render_pixmap (int cx, int cy, double zoom, int width, int height, int dept
   region.Y1 = MAX (0, MIN (PCB->MaxHeight, region.Y1));
   region.Y2 = MAX (0, MIN (PCB->MaxHeight, region.Y2));
 
-  hid_expose_callback (&ghid_hid, &ghid_graphics, &region, NULL);
+  hid_expose_callback (&ghid_graphics, &region, NULL);
   hidgl_flush_triangles (priv->hidgl);
   glPopMatrix ();
 
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index db15996..1a8d9f1 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -831,7 +831,7 @@ Benchmark (int argc, char **argv, Coord x, Coord y)
   do
     {
       XFillRectangle (display, pixmap, bg_gc, 0, 0, view_width, view_height);
-      hid_expose_callback (&lesstif_hid, &lesstif_graphics, &region, 0);
+      hid_expose_callback (&lesstif_graphics, &region, 0);
       XSync (display, 0);
       time (&end);
       i++;
@@ -2579,7 +2579,7 @@ idle_proc (XtPointer dummy)
 	    }
 	}
       DrawBackgroundImage();
-      hid_expose_callback (&lesstif_hid, &lesstif_graphics, &region, 0);
+      hid_expose_callback (&lesstif_graphics, &region, 0);
       draw_grid ();
       lesstif_use_mask (HID_MASK_OFF);
       show_crosshair (0); /* To keep the drawn / not drawn info correct */
@@ -3803,7 +3803,7 @@ pinout_callback (Widget da, PinoutData * pd,
   region.Y2 = PCB->MaxHeight;
 
   XFillRectangle (display, pixmap, bg_gc, 0, 0, pd->v_width, pd->v_height);
-  hid_expose_callback (&lesstif_hid, &lesstif_graphics, &region, pd->item);
+  hid_expose_callback (&lesstif_graphics, &region, pd->item);
 
   pinout = 0;
   view_left_x = save_vx;
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index d816497..b0f7aff 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -649,7 +649,7 @@ nelma_start_png_export()
 	linewidth = -1;
 	lastbrush = (gdImagePtr)((void *) -1);
 
-	hid_expose_callback(&nelma_hid, &nelma_graphics, &region, 0);
+	hid_expose_callback(&nelma_graphics, &region, 0);
 }
 
 static void 
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index e00aef0..8995153 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -655,7 +655,7 @@ png_hid_export_to_file (FILE * the_file, HID_Attr_Val * options)
 	}
     }
 
-  hid_expose_callback (&png_hid, &png_graphics, bounds, 0);
+  hid_expose_callback (&png_graphics, bounds, 0);
 
   memcpy (LayerStack, saved_layer_stack, sizeof (LayerStack));
   PCB->Flags = save_flags;
diff --git a/src/hid/ps/eps.c b/src/hid/ps/eps.c
index 832fdf1..fc95206 100644
--- a/src/hid/ps/eps.c
+++ b/src/hid/ps/eps.c
@@ -303,7 +303,7 @@ eps_hid_export_to_file (FILE * the_file, HID_Attr_Val * options)
   fprintf (f,
 	   "/a { gsave setlinewidth translate scale 0 0 1 5 3 roll arc stroke grestore} bind def\n");
 
-  hid_expose_callback (&eps_hid, &eps_graphics, bounds, 0);
+  hid_expose_callback (&eps_graphics, bounds, 0);
 
   fprintf (f, "showpage\n");
 
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index 3b6c0e9..cf41508 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -693,13 +693,13 @@ ps_hid_export_to_file (FILE * the_file, HID_Attr_Val * options)
 
       global.doing_toc = 1;
       global.pagecount = 1;  /* 'pagecount' is modified by hid_expose_callback() call */
-      hid_expose_callback (&ps_hid, &ps_graphics, &global.region, 0);
+      hid_expose_callback (&ps_graphics, &global.region, 0);
     }
 
   global.pagecount = 1; /* Reset 'pagecount' if single file */
   global.doing_toc = 0;
   ps_set_layer (NULL, 0, -1);  /* reset static vars */
-  hid_expose_callback (&ps_hid, &ps_graphics, &global.region, 0);
+  hid_expose_callback (&ps_graphics, &global.region, 0);
 
   if (the_file)
     fprintf (the_file, "showpage\n");
